mw-webapp/src/model/businessModel/User.ts

```diff
@@ -282,21 +361,105 @@ export class User {
    */
   public wayRequests: WayNotSaturatedUser[];
 
-  constructor(userData: User) {
+  constructor(userData: UserProps) {
     makeAutoObservable(this);
     this.uuid = userData.uuid;
     this.name = userData.name;
     this.email = userData.email;
     this.description = userData.description;
     this.createdAt = userData.createdAt;
-    this.customWayCollections = userData.customWayCollections;
-    this.defaultWayCollections = userData.defaultWayCollections;
+    this.customWayCollections = userData.customWayCollections.map(customWayCollection => new WayCollection(customWayCollection));
+    this.defaultWayCollections = new DefaultWayCollections(userData.defaultWayCollections);
     this.favoriteForUserUuids = userData.favoriteForUserUuids;
     this.favoriteUsers = userData.favoriteUsers;
-    this.tags = userData.tags;
+    this.tags = userData.tags.map(tag => new UserTag(tag));
     this.imageUrl = userData.imageUrl;
     this.isMentor = userData.isMentor;
     this.wayRequests = userData.wayRequests;
   }
 
+  /**
+   * Update user's name
+   */
+  public updateName(nameToUpdate: string): void {
+    this.name = nameToUpdate;
+    const ways = this.defaultWayCollections.own.ways.map((way) => {
+      const owner = new UserPreviewShort({...way.owner, name: nameToUpdate});
+
+      return new WayPreview({
+        ...way,
+        owner,
+      });
+    });
+    const ownCollection = new WayCollection({...this.defaultWayCollections.own, ways});
+
+    this.defaultWayCollections = new DefaultWayCollections({
+      ...this.defaultWayCollections,
+      own: ownCollection,
+    });
+  }
+
+  /**
+   * Update user's description
+   */
+  public updateDescription(descriptionToUpdate: string): void {
+    this.description = descriptionToUpdate;
+  }
+
+  /**
+   * Update isMentor property
+   */
+  public updateIsMentor(isMentorToUpdate: boolean): void {
+    this.isMentor = isMentorToUpdate;
+    const ways = this.defaultWayCollections.own.ways.map((way) => {
+      const owner = new UserPreviewShort({...way.owner, isMentor: isMentorToUpdate});
+
+      return new WayPreview({
+        ...way,
+        owner,
+      });
+    });
+    const ownCollection = new WayCollection({...this.defaultWayCollections.own, ways});
+
+    this.defaultWayCollections = new DefaultWayCollections({
+      ...this.defaultWayCollections,
+      own: ownCollection,
+    });
```

Comment:
The code in `updateName` and `updateIsMentor` methods is almost identical. It can be fixed with additional private method. 
```ts
private updateDefaultWayCollections(updateKey: string, updateValue: string | boolean) {
 const ways = this.defaultWayCollections.own.ways.map((way) => {
      const owner = new UserPreviewShort({...way.owner, [updateKey]: updateValue});

      return new WayPreview({
        ...way,
        owner,
      });
    });
    const ownCollection = new WayCollection({...this.defaultWayCollections.own, ways});

    this.defaultWayCollections = new DefaultWayCollections({
      ...this.defaultWayCollections,
      own: ownCollection,
    });
}
... 
public updateName(nameToUpdate: string): void {
    this.name = nameToUpdate;
    updateDefaultWayCollections("name", nameToUpdate);
}
...
 public updateIsMentor(isMentorToUpdate: boolean): void {
    this.isMentor = isMentorToUpdate;
    updateDefaultWayCollections("isMentor", isMentorToUpdate);
}
```

Reply:
Interesting idea